### PR TITLE
[index] Remove Boost.Move and Boost.Preprocessor emulation...

### PR DESCRIPTION
**Pull request by awulkiew**
...of move semantics and variadic templates

### CODE REVIEW

**Review by barendgehrels — APPROVED** [review]
Looks good to me

File: include/boost/geometry/index/detail/varray.hpp [hunk 1]
```diff
@@ -661,14 +644,14 @@ class varray
     //!
     //! @par Complexity
     //!   Constant O(1).
-    void push_back(BOOST_RV_REF(value_type) value)
+    void push_back(value_type && value)
```
Comment (vissarion) [inline include/boost/geometry/index/detail/varray.hpp:647]: I think you should avoid space before `&&` to be consistent. Also in other places in this PR.

File: include/boost/geometry/index/detail/varray.hpp [hunk 2]
```diff
@@ -725,7 +708,7 @@ class varray
         {
             // TODO - should move be used only if it's nonthrowing?
             value_type & r = *(this->end() - 1);
-            sv::construct(dti(), this->end(), boost::move(r));                      // may throw
+            sv::construct(dti(), this->end(), std::move(r));                        // may throw
```
Comment (vissarion) [inline include/boost/geometry/index/detail/varray.hpp:711]: This line introduces a trailing whitespace.

Comment (vissarion) [reply include/boost/geometry/index/detail/varray.hpp:711]: Also happens in other places in this PR.

**Review by vissarion — APPROVED** [review]
LGTM, thanks!

**Pull request merged**